src/nixos.rs

```diff
@@ -173,11 +204,26 @@ impl OsRebuildArgs {
     }
 }
 
-pub fn toplevel_for<S: AsRef<str>>(hostname: S, installable: Installable) -> Installable {
+pub fn get_final_attr(build_vm: bool, with_bootloader: bool) -> String {
+    let attr = if build_vm && with_bootloader {
+        "vmWithBootLoader"
+    } else if build_vm {
+        "vm"
+    } else {
+        "toplevel"
+    };
+    String::from(attr)
+}
+
+pub fn toplevel_for<S: AsRef<str>>(
+    hostname: S,
+    installable: Installable,
+    final_attr: String,
+) -> Installable {
     let mut res = installable.clone();
     let hostname = hostname.as_ref().to_owned();
 
-    let toplevel = ["config", "system", "build", "toplevel"]
+    let toplevel = ["config", "system", "build", &final_attr]
         .into_iter()
         .map(String::from);
```

Comment:
```diff
-    final_attr: String,
+    final_attr: &str,
 ) -> Installable {
     let mut res = installable.clone();
     let hostname = hostname.as_ref().to_owned();

-    let toplevel = ["config", "system", "build", &final_attr]
+    let toplevel = ["config", "system", "build", final_attr]
         .into_iter()
         .map(String::from);
```

Passing `final_attr` as a string slice (`&str`) ensures it is immutable and avoids unnecessary allocations.